### PR TITLE
[INLONG-4875][Manager] Re-issue failed source task

### LINKED ISSUE #4875: [Bug][Manager] Agent task did not retry after the incorrect task config is fixed
### What happened

1. Create a binlog task with a incorrect database host ip
2. The agent reports task failure
3. Fix the task config through the manager
4. The agent won't retry and the task still remains failure.

### What you expected to happen

The task status becomes normal after fixing.

### How to reproduce

See "what happened"

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
Re-issue failed source task when configuration is fixed.

- Fixes #4875

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java [hunk 1]
```diff
@@ -160,6 +160,14 @@ public void updateOpt(SourceRequest request, Integer groupStatus, String operato
         entity.setVersion(entity.getVersion() + 1);
         entity.setModifier(operator);
         entity.setModifyTime(new Date());
+
+        // Re-issue task if necessary
+        if (GroupStatus.forCode(groupStatus).equals(GroupStatus.CONFIG_SUCCESSFUL)) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:165]: What if the GroupStatus is failed?

Comment (woofyzhao) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:166]: Then we need to reset status to NEW. Fixed.

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java [hunk 2]
```diff
@@ -162,10 +162,11 @@ public void updateOpt(SourceRequest request, Integer groupStatus, String operato
         entity.setModifyTime(new Date());
 
         // Re-issue task if necessary
+        entity.setPreviousStatus(entity.getStatus());
         if (GroupStatus.forCode(groupStatus).equals(GroupStatus.CONFIG_SUCCESSFUL)) {
-            LOGGER.info("re-issuing source task {}, previous status = {}", entity.getId(), entity.getStatus());
-            entity.setPreviousStatus(entity.getStatus());
             entity.setStatus(SourceStatus.TO_BE_ISSUED_ADD.getCode());
+        } else {
+            entity.setStatus(SourceStatus.SOURCE_NEW.getCode());
```
Comment (gong) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:169]: Not `CONFIG_SUCCESSFUL`, code will set `SourceStatus.SOURCE_NEW`.   Do we need  set `SourceStatus.SOURCE_NEW` by judging status?

Comment (healchow) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java:169]: Do you have any ideas?

Comment (woofyzhao) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java]: Limit the original source status range to be either success or fail. Other status leave unchanged.

**Review by gong — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**